tile-window.lisp

```diff
@@ -336,6 +339,47 @@ when selecting another window."
     (when pulled-window
       (pull-window pulled-window))))
 
+(defun pull-selected-window (menu)
+  (let ((selected-window (second
+                          (nth (menu-state-selected menu)
+                               (menu-state-table menu)))))
+    (unless (some (lambda (frame)
+                    (= (window-id (frame-window (if (listp frame) (first frame) frame)))
+                       (window-id selected-window)))
+                  (tile-group-frame-tree (current-group)))
+      (pull-window selected-window))))
+
+(defun restored-windows (pulled-window original-group-windows)
+  (let ((new-group-windows (group-windows (current-group))))
+    (if pulled-window
+        (append (list pulled-window)
+                (remove pulled-window
+                        (mapcar (lambda (window)
+                                  (find-if
+                                   (lambda (w)
+                                     (eq (window-id w) (window-id window)))
+                                   new-group-windows))
+                                original-group-windows)))
+        original-group-windows)))
+
+(add-hook *menu-selection-hook*
+          (lambda (menu)
+            (when *is-in-visual-pull*
+              (pull-selected-window menu))))
+
+(defcommand (visual-pull-from-windowlist tile-group) () ()
+  (setf *is-in-visual-pull* t)
```

Comment:
Wouldn't it be better to do
(let ((*is-in-visual-pull* t))
  (declare (special **is-in-visual-pull*))
  ...)

Reply:
This would be a better idea, but we need to decide on what to do w.r.t. the external hook being non-NIL first